frame/nfts/src/lib.rs

```diff
@@ -0,0 +1,1723 @@
+		_,
+		(
+			NMapKey<Blake2_128Concat, T::AccountId>, // owner
+			NMapKey<Blake2_128Concat, T::CollectionId>,
+			NMapKey<Blake2_128Concat, T::ItemId>,
+		),
+		(),
+		OptionQuery,
+	>;
+
+	/// The collections owned by any given account; set out this way so that collections owned by
+	/// a single account can be enumerated.
+	#[pallet::storage]
+	pub(super) type CollectionAccount<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::AccountId,
+		Blake2_128Concat,
+		T::CollectionId,
+		(),
+		OptionQuery,
+	>;
+
+	/// The items in existence and their ownership details.
+	#[pallet::storage]
+	/// Stores collection roles as per account.
+	pub(super) type CollectionRoleOf<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::CollectionId,
+		Blake2_128Concat,
+		T::AccountId,
+		CollectionRoles,
+		OptionQuery,
+	>;
+
+	/// The items in existence and their ownership details.
+	#[pallet::storage]
+	pub(super) type Item<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::CollectionId,
+		Blake2_128Concat,
+		T::ItemId,
+		ItemDetails<T::AccountId, ItemDepositOf<T, I>, ApprovalsOf<T, I>>,
+		OptionQuery,
+	>;
+
+	/// Metadata of a collection.
+	#[pallet::storage]
+	pub(super) type CollectionMetadataOf<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Blake2_128Concat,
+		T::CollectionId,
+		CollectionMetadata<DepositBalanceOf<T, I>, T::StringLimit>,
+		OptionQuery,
+	>;
+
+	/// Metadata of an item.
+	#[pallet::storage]
+	pub(super) type ItemMetadataOf<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::CollectionId,
+		Blake2_128Concat,
+		T::ItemId,
+		ItemMetadata<DepositBalanceOf<T, I>, T::StringLimit>,
+		OptionQuery,
+	>;
+
+	/// Attributes of a collection.
+	#[pallet::storage]
+	pub(super) type Attribute<T: Config<I>, I: 'static = ()> = StorageNMap<
+		_,
+		(
+			NMapKey<Blake2_128Concat, T::CollectionId>,
+			NMapKey<Blake2_128Concat, Option<T::ItemId>>,
+			NMapKey<Blake2_128Concat, AttributeNamespace<T::AccountId>>,
+			NMapKey<Blake2_128Concat, BoundedVec<u8, T::KeyLimit>>,
+		),
+		(BoundedVec<u8, T::ValueLimit>, AttributeDepositOf<T, I>),
+		OptionQuery,
+	>;
+
+	/// A price of an item.
+	#[pallet::storage]
+	pub(super) type ItemPriceOf<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::CollectionId,
+		Blake2_128Concat,
+		T::ItemId,
+		(ItemPrice<T, I>, Option<T::AccountId>),
+		OptionQuery,
+	>;
+
+	/// Item attribute approvals.
+	#[pallet::storage]
+	pub(super) type ItemAttributesApprovalsOf<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::CollectionId,
+		Blake2_128Concat,
+		T::ItemId,
+		ItemAttributesApprovals<T, I>,
+		ValueQuery,
+	>;
+
+	/// Stores the `CollectionId` that is going to be used for the next collection.
+	/// This gets incremented by 1 whenever a new collection is created.
```

Comment:
It depends on how the `Incremented` trait is implemented as to how many it gets incremented by.

```suggestion
	/// This gets incremented whenever a new collection is created.
```